### PR TITLE
refs#24012 create default meeting agenda

### PR DESCRIPTION
**Pull request by iuga-paula**

### CODE REVIEW

**Review by mxrtinax — APPROVED** [review]

File: modules/edw_event/modules/edw_event_agenda/edw_event_agenda.services.yml [hunk 1]
```diff
@@ -8,3 +8,9 @@ services:
     arguments: [ '@entity_type.manager' ]
     tags:
       - { name: event_subscriber }
+
+  edw_event.meeting.commands:
```
Comment (Anthelixes) [inline modules/edw_event/modules/edw_event_agenda/edw_event_agenda.services.yml:12]: Please rename to: `edw_event_agenda.meeting.agenda.commands`

File: modules/edw_event/modules/edw_event_agenda/src/Commands/AgendaCommands.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Drupal\edw_event_agenda\Commands;
+
+use Drupal\edw_event\Services\Importer\MeetingSectionImporter;
+use Drupal\edw_event_agenda\Services\MeetingAgendaService;
+use Drush\Commands\DrushCommands;
+
+/**
+ * Class MeetingsCommands for meetings and meeting sections.
```
Comment (Anthelixes) [inline modules/edw_event/modules/edw_event_agenda/src/Commands/AgendaCommands.php:10]: Please update the description. "Class AgendaCommands for agenda items." or similar.

File: modules/edw_event/modules/edw_event_agenda/src/Commands/AgendaCommands.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+class AgendaCommands extends DrushCommands {
+  /**
+   * The MeetingAgenda service.
+   *
+   * @var \Drupal\edw_event_agenda\Services\MeetingAgendaService
+   */
+  private MeetingAgendaService $meetingAgendaService;
+
+  /**
+   * Constructs a new MeetingsCommands object.
```
Comment (Anthelixes) [inline modules/edw_event/modules/edw_event_agenda/src/Commands/AgendaCommands.php:21]: Update the description.

File: modules/edw_event/modules/edw_event_agenda/src/Commands/AgendaCommands.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+   * @param \Drupal\edw_event_agenda\Services\MeetingAgendaService $meetingAgendaService
+   *   The database connection.
+   */
+  public function __construct(MeetingAgendaService $meetingAgendaService) {
+    parent::__construct();
+    $this->meetingAgendaService = $meetingAgendaService;
+  }
+
+  /**
+   * Create a meeting section with phase.
```
Comment (Anthelixes) [inline modules/edw_event/modules/edw_event_agenda/src/Commands/AgendaCommands.php:32]: Update the description.

File: modules/edw_event/modules/edw_event_agenda/src/Services/MeetingAgendaService.php [hunk 5]
```diff
@@ -113,4 +111,48 @@ public function loadMultipleAgendaItems(array $ids) {
...
+      ->condition('field_is_default_agenda', TRUE)
+      ->execute();
+    $defaultAgenda = reset($defaultAgenda);
+
+    if (!empty($defaultAgenda)) {
+      return;
+    }
+
+    // If there is a non-default agenda with that name make it default.
+    $properties = [
```
Comment (Anthelixes) [inline modules/edw_event/modules/edw_event_agenda/src/Services/MeetingAgendaService.php:137]: Existing agenda items should not be changed. Return instead a message to inform the user.

File: modules/edw_event/modules/edw_event_agenda/src/Services/MeetingAgendaService.php [hunk 6]
```diff
@@ -113,4 +111,48 @@ public function loadMultipleAgendaItems(array $ids) {
...
+      $term->set('field_is_default_agenda', TRUE);
+      $term->save();
+      return;
+    }
+
+
+    // Otherwise, create a new default agenda.
+    $properties['field_is_default_agenda'] = TRUE;
+    $defaultAgenda = $this->termStorage->create($properties);
+    $defaultAgenda->save();
```
Comment (Anthelixes) [inline modules/edw_event/modules/edw_event_agenda/src/Services/MeetingAgendaService.php:155]: Display a message when the term is created. (e.g.: A new agenda item with id x was created.)

**Review by Anthelixes — APPROVED** [review]

**Pull request merged**